spring-integration-http/src/main/java/org/springframework/integration/http/outbound/AsyncHttpRequestExecutingMessageHandler.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.http.outbound;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.AsyncClientHttpRequestFactory;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.util.Assert;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+import org.springframework.util.concurrent.SettableListenableFuture;
+import org.springframework.web.client.AsyncRestTemplate;
+import org.springframework.web.client.ResponseErrorHandler;
+
+import java.net.URI;
+import java.util.List;
+
+/**
+ * A {@link MessageHandler} implementation that executes HTTP requests by delegating
+ * to a {@link AsyncRestTemplate} instance.
+ * @see HttpRequestExecutingMessageHandler
+ * @author Shiliang Li
+ * @since 5.0
+ */
+public class AsyncHttpRequestExecutingMessageHandler extends AbstractHttpRequestExecutingMessageHandler {
```

Comment:
I do override the getComponentType(), see line 104~107 in AsyncHttpRequestExecutingMessageHandler.java